### PR TITLE
proto: compatible with proto3.

### PR DESCRIPTION
**Pull request by disksing**
Suppress errors when a newer version of tikv (probably v1.1) connects to pd v1.0.x.
Related issue: https://github.com/pingcap/tikv/issues/2681

### CODE REVIEW

File: server/grpc_service.go [hunk 1]
```diff
@@ -389,8 +389,8 @@ func (s *Server) AskSplit(ctx context.Context, request *pdpb.AskSplitRequest) (*
 	if cluster == nil {
 		return &pdpb.AskSplitResponse{Header: s.notBootstrappedHeader()}, nil
 	}
-	if request.GetRegion().GetStartKey() == nil {
-		return nil, errors.New("missing region start key for split")
+	if request.GetRegion() == nil {
```
Comment (BusyJay) [inline server/grpc_service.go]: Seems no test covers it.

**Comment by BusyJay** [comment]
LGTM

**Comment by disksing** [comment]
/run-all-tests tidb=release-1.0 tikv=release-1.0 tidb-test=release-1.0

**Comment by disksing** [comment]
PTAL @Connor1996 @siddontang

**Pull request merged**